packages/ubuntu_desktop_installer/lib/pages/connect_to_internet/wifi_model.dart

```diff
@@ -0,0 +1,314 @@
+  void selectDevice(WifiDevice? device) {
+    selectedDevice?.removeListener(notifyListeners);
+    device?.addListener(notifyListeners);
+    _selected = device;
+    notifyListeners();
+  }
+
+  Future requestScan({String? ssid}) async {
+    if (!isEnabled) return;
+    final scans = <Future<void>>[];
+    for (final device in devices) {
+      if (device.isAvailable) {
+        scans.add(device.requestScan(ssid: ssid));
+      }
+    }
+    return Future.wait(scans);
+  }
+}
+
+class WifiDevice extends NetworkDevice {
+  WifiDevice(NetworkManagerDevice device, [UdevService? udev])
+      : _wireless = device.wireless!,
+        super(device, udev) {
+    addProperties(_wireless.propertiesChanged);
+    addPropertyListener('AccessPoints', _updateAccessPoints);
+    addPropertyListener('ActiveAccessPoint', notifyListeners);
+    addPropertyListener('LastScan', () {
+      _setLastScan(device.wireless!.lastScan);
+      _setScanning(false);
+      if (_completer?.isCompleted == false) {
+        _completer!.complete(device);
+      }
+    });
+  }
+
+  void init() => selectAccessPoint(activeAccessPoint);
+
+  @override
+  void dispose() {
+    _resetAccessPoints();
+    super.dispose();
+  }
+
+  @override
+  bool get isActive => super.isActive && activeAccessPoint != null;
+
+  bool get canConnect =>
+      isActive &&
+      _selected != null &&
+      _selected!.name == activeAccessPoint?.name;
+
+  final NetworkManagerDeviceWireless _wireless;
+
+  List<AccessPoint>? _accessPoints;
+  List<AccessPoint> get accessPoints => _accessPoints ??= _getAccessPoints();
+
+  List<AccessPoint> _getAccessPoints() {
+    final ssids = <List<int>>[];
+
+    bool hasSsid(List<int> ssid) {
+      return ssids.any((s) => const ListEquality().equals(s, ssid));
+    }
+
+    bool acceptSsid(List<int> ssid) {
+      if (ssid.isEmpty || hasSsid(ssid)) {
+        return false;
+      }
+      ssids.add(ssid);
+      return true;
+    }
+
+    final accessPoints = _wireless.accessPoints
+        .where((ap) => acceptSsid(ap.ssid))
+        .map((ap) => AccessPoint(ap))
+        .sorted((a, b) => b.strength.compareTo(a.strength));
+    log.debug(() =>
+        'Update access points: ${describeAccessPoints(accessPoints)} (device: $this)');
+    return accessPoints;
+  }
+
+  void _resetAccessPoints() {
+    if (_accessPoints == null) return;
+    for (final ap in _accessPoints!) {
+      ap.dispose();
+    }
+    _accessPoints = null;
+  }
+
+  void _updateAccessPoints() {
+    _resetAccessPoints();
+    notifyListeners();
+  }
+
+  AccessPoint? get activeAccessPoint {
+    if (_wireless.activeAccessPoint == null) return null;
+    return accessPoints.firstWhereOrNull((ap) {
+      return ap.accessPoint == _wireless.activeAccessPoint;
+    });
+  }
+
+  bool isActiveAccessPoint(AccessPoint accessPoint) {
+    return accessPoint._accessPoint == _wireless.activeAccessPoint;
+  }
+
+  AccessPoint? _selected;
```

Comment:
Same remark as for `WifiModel`: there are a bunch of references to `_selected` in getters and methods which don't change its value, for consistency and improved readability they should probably be replaced by `selectedAccessPoint`.